pkg/kn/flags/podspec_helper.go

```diff
@@ -255,6 +258,14 @@ func UpdateImagePullSecrets(spec *corev1.PodSpec, pullsecrets string) {
 	}
 }
 
+// UpdateContainers updates the containers array with additional ones provided from file or os.Stdin
+func UpdateContainers(spec *corev1.PodSpec, containers []corev1.Container) {
+	c := containerOfPodSpec(spec)
+	spec.Containers = []corev1.Container{}
+	spec.Containers = append(spec.Containers, *c)
+	spec.Containers = append(spec.Containers, containers...)
```

Comment:
I haven't dived deep enough into the code, but shouldn't you correlated via the container name instead of just appending ? I.e. lookup whether the extra containers' name are already present in the existing containers and replace those in case instead of just appending ?

Reply:
Currently we're in `drop&replace` as I was aiming to have a solution to replace all sidecars at once. E.g. when `Ksvc` is created from file without named containers etc.


Is it better now? I can confirm that `kn apply` works with the change.
https://github.com/knative/client/pull/1382/commits/21567d37634c3eaface64afb2464e92e68a2d308